ding/reward_model/drex_reward_model.py

```diff
@@ -0,0 +1,206 @@
+        num_bins = len(self.created_data)
+        assert num_bins >= 2
+
+        # add full trajs (for use on Enduro)
+        si = np.random.randint(6, size=num_trajs)
+        sj = np.random.randint(6, size=num_trajs)
+        step = np.random.randint(3, 7, size=num_trajs)
+        for n in range(num_trajs):
+            # pick two random demonstrations
+            bi, bj = np.random.choice(num_bins, size=(2, ), replace=False)
+            ti = np.random.choice(len(self.created_data[bi]))
+            tj = np.random.choice(len(self.created_data[bj]))
+            # create random partial trajs by finding random start frame and random skip frame
+            traj_i = self.created_data[bi][ti][si[n]::step[n]]  # slice(start,stop,step)
+            traj_j = self.created_data[bj][tj][sj[n]::step[n]]
+
+            label = int(bi <= bj)
+
+            self.training_obs.append((traj_i, traj_j))
+            self.training_labels.append(label)
+            max_traj_length = max(max_traj_length, len(traj_i), len(traj_j))
+
+        # fixed size snippets with progress prior
+        rand_length = np.random.randint(min_snippet_length, max_snippet_length, size=num_snippets)
+        for n in range(num_snippets):
+            # pick two random demonstrations
+            bi, bj = np.random.choice(num_bins, size=(2, ), replace=False)
+            ti = np.random.choice(len(self.created_data[bi]))
+            tj = np.random.choice(len(self.created_data[bj]))
+            # create random snippets
+            # find min length of both demos to ensure we can pick a demo no earlier
+            # than that chosen in worse preferred demo
+            min_length = min(len(self.created_data[bi][ti]), len(self.created_data[bj][tj]))
+            if bi < bj:  # pick tj snippet to be later than ti
+                ti_start = np.random.randint(min_length - rand_length[n] + 1)
+                # print(ti_start, len(demonstrations[tj]))
+                tj_start = np.random.randint(ti_start, len(self.created_data[bj][tj]) - rand_length[n] + 1)
+            else:  # ti is better so pick later snippet in ti
+                tj_start = np.random.randint(min_length - rand_length[n] + 1)
+                # print(tj_start, len(demonstrations[ti]))
+                ti_start = np.random.randint(tj_start, len(self.created_data[bi][ti]) - rand_length[n] + 1)
+            # skip everyother framestack to reduce size
+            traj_i = self.created_data[bi][ti][ti_start:ti_start + rand_length[n]:2]
+            traj_j = self.created_data[bj][tj][tj_start:tj_start + rand_length[n]:2]
+
+            max_traj_length = max(max_traj_length, len(traj_i), len(traj_j))
+            label = int(bi <= bj)
+            self.training_obs.append((traj_i, traj_j))
+            self.training_labels.append(label)
+        self._logger.info(("maximum traj length: {}".format(max_traj_length)))
+        return self.training_obs, self.training_labels
+
+    def train(self):
```

Comment:
what is difference of train method between trex and drex, I think there should be no difference between them. As for implementation like `pred_data` method, you can also add this method in trex to unify them.